### PR TITLE
fix: unexpected chinese punctuation

### PR DESCRIPTION
**Pull request by ufolux**
*Fixed the issue with abnormal Chinese punctuation marks.*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by goderbauer** [comment]
Hello @ufolux, thanks for the contribution. Can you please sign the CLA as instructed by the previous comment? Also, it looks like this change is failing the analyzer check. Can you please take a look and fix that? Thank you!

**Comment by ufolux** [comment]
> Hello @ufolux, thanks for the contribution. Can you please sign the CLA as instructed by the previous comment? Also, it looks like this change is failing the analyzer check. Can you please take a look and fix that? Thank you!

Yes, I already signed CLA. I'll take a look on the check.

**Comment by ufolux** [comment]
> Hello @ufolux, thanks for the contribution. Can you please sign the CLA as instructed by the previous comment? Also, it looks like this change is failing the analyzer check. Can you please take a look and fix that? Thank you!

Hi @goderbauer , I don't understand what is the reason for the failure of the analyzer check because of the output is like gibberish. Can you help me on this? I just made several changes on some strings. Thank you!

File: packages/flutter/lib/src/widgets/basic.dart [hunk 1]
```diff
@@ -390,7 +390,7 @@ class Opacity extends SingleChildRenderObjectWidget {
 ///     ).createShader(bounds);
 ///   },
 ///   child: const Text(
-///     'I’m burning the memories',
+///     'I'm burning the memories',
```
Comment (goderbauer) [inline packages/flutter/lib/src/widgets/basic.dart:393]: If you want to use `'` in a string you have to either escape it like `\'` or use `"` as a string delimiter:
```suggestion
///     "I'm burning the memories",
```

File: packages/flutter/lib/src/widgets/basic.dart [hunk 2]
```diff
@@ -5034,8 +5034,8 @@ class Row extends Flex {
 ///     const Text('Through the night, we have one shot to live another day'),
 ///     const Text('We cannot let a stray gunshot give us away'),
 ///     const Text('We will fight up close, seize the moment and stay in it'),
-///     const Text('It’s either that or meet the business end of a bayonet'),
-///     const Text('The code word is ‘Rochambeau,’ dig me?'),
+///     const Text('It's either that or meet the business end of a bayonet'),
+///     const Text('The code word is 'Rochambeau,' dig me?'),
```
Comment (goderbauer) [inline packages/flutter/lib/src/widgets/basic.dart:5038]: same here:

```suggestion
///     const Text("It's either that or meet the business end of a bayonet"),
///     const Text("The code word is 'Rochambeau,' dig me?"),
```

**Review by ufolux — COMMENTED** [review]
@goderbauer Thank you for the heads up! Just corrected escape.

**Review by goderbauer — APPROVED** [review]
LGTM

**Review by Piinks — APPROVED** [review]
Welcome @ufolux! Thank you, LGTM! 🎉

**Comment by ufolux** [comment]
Thank you guys!

**Pull request merged**